server/datastore/mysql/software.go

```diff
@@ -3679,6 +3682,14 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 				}
 			}
 		}
+		for vppAppAdamID, software := range byVPPAdamID {
+			if software.VPPAppSelfService != nil && *software.VPPAppSelfService {
+				if filteredByVPPAdamID[vppAppAdamID] == nil {
+					// remove the software title from byVPPAdamID
+					delete(byVPPAdamID, vppAppAdamID)
+				}
+			}
+		}
```

Comment:
This is such a small diff because software titles are handled above. Expand the compressed code:
```
	if opts.SelfServiceOnly {
		for _, software := range bySoftwareTitleID {
			if software.PackageSelfService != nil && *software.PackageSelfService {
				if filteredBySoftwareTitleID[software.ID] == nil {
					// remove the software title from bySoftwareTitleID
					delete(bySoftwareTitleID, software.ID)
				}
			}
		}
		for vppAppAdamID, software := range byVPPAdamID {
			if software.VPPAppSelfService != nil && *software.VPPAppSelfService {
				if filteredByVPPAdamID[vppAppAdamID] == nil {
					// remove the software title from byVPPAdamID
					delete(byVPPAdamID, vppAppAdamID)
				}
			}
		}
```